fahrschein/src/main/java/org/zalando/fahrschein/CursorOffsetCommitException.java

```diff
@@ -0,0 +1,61 @@
+package org.zalando.fahrschein;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import org.zalando.fahrschein.domain.Cursor;
+
+@SuppressWarnings("serial")
+public class CursorOffsetCommitException extends IOException {
+    private final int statusCode;
+    private final Cursor cursor;
+    private final String subscriptionId;
+
+    public CursorOffsetCommitException(int statusCode, Cursor cursor, String subscriptionId, IOException cause) {
+        super(formatMessage(statusCode, cursor, subscriptionId), cause);
+        this.statusCode = statusCode;
+        this.cursor = cursor;
+        this.subscriptionId = subscriptionId;
+    }
+
+    public CursorOffsetCommitException(int statusCode, Cursor cursor, String subscriptionId) {
+        super(formatMessage(statusCode, cursor, subscriptionId));
+        this.statusCode = statusCode;
+        this.cursor = cursor;
+        this.subscriptionId = subscriptionId;
+    }
+
+    private static String formatMessage(int statusCode, Cursor cursor, String subscriptionId) {
+        String msg;
+        switch (statusCode) {
+            case 422:
+                msg = String.format(
+                    "Cursor for subscription [%s] to event [%s] in partition [%s] with offset [%s] failed to commit because of error 422 (Unprocessable Entity). " +
```

Comment:
```suggestion
                    "Cursor for subscription [%s] and event type [%s] in partition [%s] with offset [%s] failed to commit because of error 422 (Unprocessable Entity). " +
```